ptero_workflow/implementation/backend.py

```diff
@@ -216,6 +216,47 @@ def get_workflow_executions(self, workflow_id, since=None):
         else:
             return [], None
 
+    def get_limited_workflow_executions(self, **kwargs):
+        model_class = models.Execution
+        return self._get_limited_reports(model_class=model_class, **kwargs)
+
+    def _get_limited_reports(self, model_class, workflow_id,
+            limit, since=None):
+        query = self.session.query(model_class)
+
+        if since is not None:
+            query = query.filter(model_class.workflow_id == workflow_id,
+                            model_class.timestamp >= since)
+        else:
+            query = query.filter(model_class.workflow_id == workflow_id)
+
+        query = query.order_by(model_class.timestamp).limit(limit + 1)
+        instances = query.all()
+
+        if instances:
+            if len(instances) == limit + 1:
+                # ensure that we return exactly the right number of results
+                next_execution = instances.pop()
+            else:
+                next_execution = instances[-1]
+
+            timestamp = next_execution.timestamp
+            reports = [e.as_dict_for_limited_report()
+                    for e in instances]
+            num_remaining = self.session.query(model_class).\
+                    filter(model_class.id > instances[-1].id,
```

Comment:
While I agree this filter works in practice, we do not think of id values as members of an ordered set, so thinking of one id as being greater than another id does not make sense.  We had even considered using `id` instead of `timestamp` to drive the paging of this report but threw out that idea because any assumptions we could make about the ordering of id values was purely accidental.